### PR TITLE
Use the shrink-ray middleware to support modern compression schemes upon request.

### PR DESCRIPTION
**Pull request by Berlioz**
Thankfully, shrink-ray is implemented as a drop-in-replacement (well, it's a fork) of the compression middleware superstatic already uses.

see: https://www.npmjs.com/package/shrink-ray

e: oh nooooooo, the CI environment can't build the compression libraries involved
e2: fixed. don't merge this without making sure that it's a drop-in replacement with the origin

### CODE REVIEW

**Comment by mbleigh** [comment]
@Berlioz we should be able to get CI working (maybe we need to run some apt-gets or something before we run tests?) Can you monkey with .travis.yml and see what you can get happening?

**Comment by Berlioz** [comment]
Should be dealt with now. I think these changes are a drop-in replacement as far as the origin is concerned, but I'm spending some time to make sure.

File: lib/responder.js [hunk 1]
```diff
@@ -12,15 +12,24 @@ var mime = require('mime-types');
 var path = require('path');
 var awaitFinished = RSVP.denodeify(require('on-finished'));
 var destroy = require('destroy');
-var compress = require('compression')();
+
+// shink-ray is a fork of compression which supports advanced compression encodings but not ES5
+var legacyCompress = require('compression')();
```
Comment (mbleigh) [inline lib/responder.js]: What if we just do something like:

```js
var compressor;
try {
  compressor = require('shrink-ray')();
} catch(e) {
  compressor = require('compression')();
}
```

Then bellow you can just do:

```js
if (this.compression) {
  compressor(this.req, this.res, function() { result.stream.pipe(self.res) });
}
```

Comment (Berlioz) [inline lib/responder.js]: Good catch. That's much cleaner.

**Comment by Berlioz** [comment]
Also changed the README to reflect the new CLI options.

**Pull request merged**